azure/vnet/main.tf

```
@@ -1,34 +1,20 @@
-data "azurerm_subscription" "current" {}
-
-data "azurerm_resource_group" "network" {
-  name = var.resource_group_name
-}
-
-locals {
-  location = data.azurerm_resource_group.network.location
-  tags     = data.azurerm_resource_group.network.tags
+resource "azurerm_virtual_network" "vnet" {
+  name                = var.vnet_name
+  address_space       = var.address_space
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
```

Please consider using the data block instead of variables

```suggestion
  location            = data.azurerm_resource_group.network.location
  resource_group_name = data.azurerm_resource_group.network.name
  tags                = data.azurerm_resource_group.network.tags
```

---

this is common practice to use vars for data you define in variables.tf - data sources are used when data can be accessible within your code and you need to fetch it via the Azure API.
Use it like this make the code less readable.

---

I think both are easy to understand (if we know HCL), and the data block enforces consistency where is needed.